resolwe/process/parser.py

```diff
@@ -111,9 +113,66 @@ def __init__(self, source):
         self.source = source
         self.processes = []
         self.base_classes = set()
+        # Mapping betwwen process slug and AST node representing the class.
+        self.ast_nodes = dict()
 
         super().__init__()
 
+    def _get_called_slug(self, process_slug, called_node: ast.Call):
+        """Get the slug of the process being called.
+
+        :raises RuntimeError: when slug could not be determined (for instance when
+            the slug is being a variable).
+        """
+        # Both methods (run_process and get_latest) take argument slug as the first
+        # argument, so no separation is necessary.
+        argument_names = ("slug", "inputs")
+        arguments = [arg for arg in called_node.args]
+        all_arguments = {
+            name: value for name, value in zip_longest(argument_names, arguments)
+        }
+        all_arguments.update(
+            {keyword.arg: keyword.value for keyword in called_node.keywords}
+        )
+        assert "slug" in all_arguments, "The 'slug' is required argument"
+
+        slug = all_arguments["slug"]
+        if isinstance(slug, ast.Constant):
+            return slug.value
+        raise RuntimeError(
+            f"Spaned process slug could not be determined for the process {process_slug}."
+        )
+
+    def get_dependencies(self):
+        """Get the dependencies between the processes.
+
+        :raises RuntimeError: when dependencies could not be determined. For
+            instance when process slug is a variable.
+
+        :returns: the dictionary where key is the process slug and the
+            corresponding value is the list of processes the process can spawn.
+        """
+        slugs = dict()
+        candidate_names = (
+            "Process.get_latest",
+            "BioProcess.get_latest",
```

Comment:
This is an implementation detail of `resolwe_bio` and can be different for each package of processes.

Reply:
I used FLOW_PROCESS_RUNTIMES settings where the registered runtimes are stored.